test/helpers/adapter_helper.rb

```diff
@@ -0,0 +1,27 @@
+module Semian
+  module AdapterTest
+    include Semian::Adapter
+
+    def semian_identifier
+      :semian_adapter_test
+    end
+
+    def raw_semian_options
+      @client_options
+    end
+  end
+
+  class AdapterTestClient
+    include AdapterTest
+
+    def initialize(**args)
+      @client_options = args.merge(success_threshold: 1,
+                                   error_threshold: 1,
+                                   error_timeout: 1)
+    end
+
+    def ==(other)
+      inspect == other.inspect
```

Comment:
This still seems a little bit hacky and I don't really understand why we need this. Why inspect? Maybe `object_id == other.object_id` makes more sense?